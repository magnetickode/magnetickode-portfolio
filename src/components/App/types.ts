import { MutableRefObject } from "react";

export interface Props {
  navbarRef: MutableRefObject<HTMLDivElement | null>;
}
