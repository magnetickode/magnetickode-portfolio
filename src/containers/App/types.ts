import { ScreenSizeChangedGen, ShowStickyNavGen } from "../../store/actions/types";

export interface Props {
  isDesktop: boolean;
  themeIndex: number;
  stickyNavVisible: boolean;
  screenSizeChanged: ScreenSizeChangedGen;
  showStickyNav: ShowStickyNavGen;
}
