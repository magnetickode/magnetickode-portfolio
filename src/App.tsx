import React from "react";
import { BrowserRouter as Router } from "react-router-dom";

import Routes from "./Routes";

const App: React.FC = () => {
  return (
    <Router data-test="AppComponent">
      <div>
        <Routes />
      </div>
    </Router>
  );
};

export default App;
