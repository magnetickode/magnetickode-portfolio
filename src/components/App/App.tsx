import React from "react";
import styled, { createGlobalStyle } from "styled-components";

import Routes from "../Routes";
import Navbar from "../../containers/Navbar";
import Footer from "../../containers/Footer";
import { Theme } from "../../utils/getTheme/types";

const GlobalStyle = createGlobalStyle<{ theme: Theme }>`
  @import url('https://fonts.googleapis.com/css?family=Lato:400,700|Pacifico|Righteous&display=swap');
  html {
    height: 100%;
    font-size: 62.5%;
  }

  body {
      height: 100%;
      margin: 0;
      font-family: "Lato", sans-serif;
      font-size: 1.6rem;
      -webkit-font-smoothing: antialiased;
      -moz-osx-font-smoothing: grayscale;
      background-image: linear-gradient(
        to right,
        ${({ theme }) => theme.primaryColorRgba(0.9)}, 
        ${({ theme }) => theme.secondaryColorRgba(0.9)}),
        ${({ theme }) => theme.backgroundImage};     
      background-size: cover;
      background-position: center;
      background-attachment: fixed;
      color: ${({ theme }) => theme.textColor};
  }
`;

const StyledApp = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  margin: auto;
  overflow: hidden;
  transition: color 0.2s ease-in-out;
`;

const App: React.FC = () => (
  <>
    <GlobalStyle data-test="AppComponent" />
    <StyledApp>
      <Navbar />
      <Routes />
      <Footer />
    </StyledApp>
  </>
);

export default App;
