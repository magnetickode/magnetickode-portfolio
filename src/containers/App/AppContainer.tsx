import React, { useState } from "react";
import { ThemeProvider } from "styled-components";
import { Helmet } from "react-helmet";

import App from "../../components/App";
import AppContext from "../../appContext";
import { useStorage } from "../../hooks";
import getTheme from "../../utils/getTheme";

const AppContainer = () => {
  const [themeIndex, setThemeIndex] = useState(1);

  useStorage(themeIndex, setThemeIndex, "local", "theme");

  const theme = getTheme(themeIndex);

  const contextValue = {
    changeTheme() {
      if (themeIndex < 5) {
        setThemeIndex(themeIndex + 1);
      } else {
        setThemeIndex(0);
      }
    }
  };
  return (
    <>
      <Helmet>
        <link rel="shortcut icon" href={theme.ico.favicon} />
        <link rel="icon" href={theme.ico.size16} sizes="16x16" />
        <link rel="icon" href={theme.ico.size32} sizes="32x32" />
        <link rel="icon" href={theme.ico.size192} sizes="192x192" />
        <link rel="icon" href={theme.ico.size512} sizes="512x512" />
        <link rel="apple-touch-icon" href={theme.ico.appleTouch} />
        <link rel="manifest" href={theme.ico.manifest} />
        <meta name="theme-color" content={theme.primaryColor} />
      </Helmet>
      <AppContext.Provider value={contextValue}>
        <ThemeProvider theme={theme}>
          <App />
        </ThemeProvider>
      </AppContext.Provider>
    </>
  );
};

export default AppContainer;
