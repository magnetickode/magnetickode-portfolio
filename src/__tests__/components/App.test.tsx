import React from "react";
import { shallow, ShallowWrapper } from "enzyme";

import App from "../../components/App";

let wrapper: ShallowWrapper;

beforeEach(() => {
  wrapper = shallow(<App />);
});

it("matches snapshot", () => {
  expect(wrapper).toMatchSnapshot();
});

it("renders without error", () => {
  const AppComponent = wrapper.find("[data-test='AppComponent']");
  expect(AppComponent.length).toBe(1);
});
