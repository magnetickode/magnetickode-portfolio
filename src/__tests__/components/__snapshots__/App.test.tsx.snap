// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <GlobalStyleComponent
    data-test="AppComponent"
  />
  <BrowserRouter>
    <styled.div>
      <Connect(NavbarContainer) />
      <Routes />
      <Footer />
    </styled.div>
  </BrowserRouter>
</Fragment>
`;
